services/skus/datastore.go

```diff
@@ -32,11 +32,11 @@ const (
 	errNotFound = model.Error("not found")
 )
 
-// Datastore abstracts over the underlying datastore
+// Datastore abstracts over the underlying datastore.
 type Datastore interface {
 	datastore.Datastore
-	// CreateOrder is used to create an order for payments
-	CreateOrder(totalPrice decimal.Decimal, merchantID string, status string, currency string, location string, validFor *time.Duration, orderItems []OrderItem, allowedPaymentMethods []string) (*Order, error)
+
+	CreateOrder(ctx context.Context, dbi sqlx.ExtContext, req *model.OrderNew, items []model.OrderItem) (*model.Order, error)
```

Comment:
nitpick (non-blocking): the parameter `req` is overloaded with both http requests and `model.CreateOrderRequest`. Coupled with the [Status field](https://github.com/brave-intl/bat-go/pull/1982/files?diff=split&w=1#diff-86884800d41e3e1ec4019ea4d49e370d00ecc0b8235be9ccc8762336847e0afcR304) it a little unclear.

Its a tricky one to name, I assume the intent is to distinguish between a request to create a new order and a `model.Order` 

suggestion: what about `ordNew` or `on`

Reply:
I see your concern, it's a fair point, and I agree, it might get confusing.

The reason I chose `req` here (and has done like that in other places) is that, according to my assumption, as soon as we've crossed the boundary between transport and business logic, i.e. are in the service level and further, the word `request` looses its transport meaning. At the service level and further, I read it as "a request to create an order".

The reason I did not initially go with `orderNew` and variations on the theme is that none of them can be considered an order yet.

Additionally, I would point out that, based on the conventions established by the standard library, I would expect any mention of `http.Request` to be `r`, not `req`.

As a middle ground, and I have also done similarly in other places before, what do you think about `oreq` or `ordReq`? To me, these would mean what they are – a request to create an order.

Reply:
@clD11 Renamed it to `oreq` in the datastore interface and method, as well as in the repository it delegates to.